### PR TITLE
test: use gomock structs for testing basic grpc sync functionality

### PR DESCRIPTION
**Pull request by bacherfl**
This PR changes the `TestSync_BasicFlagSyncStates` to make use of mock implementations of the FlagSyncClient generated with gomock, to stay in line with how the tests in the `http` sync package are written.
I also modified the test to wait for the started goroutine for the sync component to finish in order to collect the correct coverage data.

### CODE REVIEW

**Comment by Kavindu-Dodan** [comment]
let's update Makefile to include this mockgen - https://github.com/open-feature/flagd/blob/main/Makefile#L58 

This helps to track and automate mockgens :)

**Review by Kavindu-Dodan — APPROVED** [review]
LGTM - Let's address the comment and other things looks good :)

File: Makefile [hunk 1]
```diff
@@ -56,7 +56,8 @@ lint:
 install-mockgen:
 	go install github.com/golang/mock/mockgen@v1.6.0
 mockgen: install-mockgen
-	mockgen -source=core/pkg/sync/http/http_sync.go -destination=core/pkg/sync/http/mock/http.go -package=syncmock
-	mockgen -source=core/pkg/eval/ievaluator.go -destination=core/pkg/eval/mock/ievaluator.go -package=evalmock
+	cd core; mockgen -source=pkg/sync/http/http_sync.go -destination=pkg/sync/http/mock/http.go -package=syncmock
+	cd core; mockgen -source=pkg/sync/grpc/grpc_sync.go -destination=pkg/sync/grpc/mock/grpc.go -package=grpcmock
```
Comment (bacherfl) [inline Makefile:60]: i had to add the `cd` into the `core` directory here due to mockgen requiring a `go.mod` file being available when generating mocks for interface aliases (i.e. the grpc client interfaces that are originally defined in another module)

Comment (Kavindu-Dodan) [reply Makefile:60]: Yes, this makes sense and is a result of the latest mono-repo refactoring :)

**Pull request merged**